schnorr_fun/src/binonce.rs

```diff
@@ -0,0 +1,170 @@
+//! Binonces for Musig and FROST Signature Schemes
+//!
+//! A binonce is a pair of public points used in the MuSig and FROST signature schemes.
+//! Your public nonces are derived from scalars which must be kept secret.
+//! Derived binonces should be unique and and must not be reused for signing under any circumstances
+//! as this can leak your secret key.
+//!
+//! Using a [`Synthetic`] nonce generator will mean you don't have to worry about passing a
+//! unique `sid` (session id) to this function for each signing session. The downside is that
+//! you must recall the result of `gen_nonces` somewhere and store it for use when you want to
+//! start the signing session.
+//!
+//! Using a [`Deterministic`] nonce generator means you **must** never start two signing
+//! sessions with nonces generated from the same `sid`. If you do your secret key will be
+//! recoverable from the two partial signatures you created with the same nonce. The upside is
+//! that you can rederive the nonces with the `sid` you originally passed
+//! to `gen_nonces` without having to store the output.
```

Comment:
Hmm but don't we have `gen_nonce` on `MuSig` and `Frost`*? Isn't that where this documentation goes? I'd expect this module to be private and to just re-export the `Nonce` and `NonceKeyPair` types from both MuSig and Frost.